### DISCUSSION TITLE
Is there a way to make only the space bar key expand and collapse the dropdown in a p-dropdown? Can I prevent enter from expanding, collapsing?

### DISCUSSION THREAD

**Discussion by kmboyle** [primefaces/community#1638] [PrimeNG] [answered]
I want to use enter for search functionality.  I have tried event.stopPropagation, event.preventDefault, keydown, binding to the show event, calling hide immediately, binding to the keydown.enter event, etc. In all cases, the dropdown still expands on enter.

**Comment by cetincakiroglu** [comment]
Hi,

Unfortunately, there isn't a feature for that. You might override the logic by accessing the component with viewChild but it's not practical and probably not worth the time.

**Reply by kmboyle** [reply, marked as answer]
Thank you for your reply @cetincakiroglu.  Thank you for mentioning viewChild. You were correct, I was able to override the logic. Here is a working demo, for anyone that's interested. [disable enter in dropdown](https://stackblitz.com/edit/gmbbnf?file=src%2Fapp%2Fdemo%2Fdropdown-basic-demo.html)
